collagraph/component.py

```diff
@@ -0,0 +1,79 @@
+from abc import ABCMeta, abstractmethod
+
+from observ import reactive, readonly
+
+
+class Component(metaclass=ABCMeta):
+    """Abstract base class for components"""
+
+    def __init__(self, props=None):
+        self._props = readonly(props)
+        self._state = reactive({})
+        self._mounted = False
+
+    @property
+    def state(self):
+        """Readonly view on local component state."""
+        return readonly(self._state)
```

Comment:
I don't really follow why this is readonly, and why you don't just store state and props on public attributes on self?